subiquity/server/controllers/filesystem.py

```diff
@@ -1264,6 +1283,60 @@ def available_target_resize_scenarios(
                 scenarios.append((vals.install_max, resize))
         return scenarios
 
+    def available_erase_install_scenarios(
+        self, install_min: int
+    ) -> list[tuple[int, GuidedStorageTargetEraseInstall]]:
+        scenarios: list[tuple[int, GuidedStorageTargetEraseInstall]] = []
+        if self.model.storage_version < 2:
+            return []
+
+        for disk in self.potential_boot_disks(check_boot=False):
+            # Skip RAID until we know how to proceed.
+            if not isinstance(disk, ModelDisk):
+                continue
+
+            for partition in disk.partitions():
+                if partition._is_in_use:
+                    continue
+
+                if partition.os is None:
+                    continue
+
+                # Make an ephemeral copy of the disk object with the relevant
+                # partition removed. Then it's as if we're installing in the
+                # resulting gap (which will include free space that was
+                # directly before or after the partition that we removed).
+                altered_disk = disk._excluding_partition(partition)
+                if not boot.can_be_boot_device(altered_disk, with_reformatting=False):
+                    continue
+
+                gap = gaps.find_gap_after_removal(
+                    altered_disk, removed_partition=partition
+                )
+                if not self.use_gap_has_enough_room_for_partitions(altered_disk, gap):
```

Comment:
Thought: Arguably this could fire a few times, is it worth adding disk/partition information to the message for better tracking in the logs?

Reply:
Good point, added!